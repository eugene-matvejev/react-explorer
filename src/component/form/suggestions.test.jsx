import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Suggestions from './suggestions';

configure({ adapter: new Adapter() });

describe('<Suggestions/>', () => {
    const props = {
        options: [
            {
                label: '{{label-only}}',
            },
            {
                label: '{{label-with-hierarchy}}',
                hierarchy: ['{{root}}', '{{sub-children}}', '{{label-with-hierarchy}}'],
            },
            {
                label: '{{label-with-description}}',
                description: '{{description}}',
            },
        ],
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<Suggestions {...props} />);

            expect(c).toMatchSnapshot();
        });


        describe('with optional props', () => {
            [
                ['data-cy', '{{data-cy}}'],
                ['className', '{{className}}'],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<Suggestions {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});
