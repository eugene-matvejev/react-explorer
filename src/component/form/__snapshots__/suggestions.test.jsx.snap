// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Suggestions/> render with default/required props 1`] = `
<div
  className="suggestions "
>
  <div
    className="suggestion undefined"
    key="0"
  >
    <div
      className="suggestion__label"
    >
      {{label-only}}
    </div>
  </div>
  <div
    className="suggestion undefined"
    key="1"
  >
    <div
      className="suggestion__hierarchy"
    >
      <span
        className="suggestion__hierarchy-label undefined"
        key="0"
      >
        {{root}}
      </span>
      <span
        className="suggestion__hierarchy-label undefined"
        key="1"
      >
        {{sub-children}}
      </span>
      <span
        className="suggestion__hierarchy-label undefined"
        key="2"
      >
        {{label-with-hierarchy}}
      </span>
    </div>
    <div
      className="suggestion__label"
    >
      {{label-with-hierarchy}}
    </div>
  </div>
  <div
    className="suggestion undefined"
    key="2"
  >
    <div
      className="suggestion__label"
    >
      {{label-with-description}}
    </div>
    <p
      className="suggestion__description"
    >
      {{description}}
    </p>
  </div>
</div>
`;

exports[`<Suggestions/> render with optional props [::className] as "{{className}}" 1`] = `
<div
  className="suggestions {{className}}"
>
  <div
    className="suggestion undefined"
    key="0"
  >
    <div
      className="suggestion__label"
    >
      {{label-only}}
    </div>
  </div>
  <div
    className="suggestion undefined"
    key="1"
  >
    <div
      className="suggestion__hierarchy"
    >
      <span
        className="suggestion__hierarchy-label undefined"
        key="0"
      >
        {{root}}
      </span>
      <span
        className="suggestion__hierarchy-label undefined"
        key="1"
      >
        {{sub-children}}
      </span>
      <span
        className="suggestion__hierarchy-label undefined"
        key="2"
      >
        {{label-with-hierarchy}}
      </span>
    </div>
    <div
      className="suggestion__label"
    >
      {{label-with-hierarchy}}
    </div>
  </div>
  <div
    className="suggestion undefined"
    key="2"
  >
    <div
      className="suggestion__label"
    >
      {{label-with-description}}
    </div>
    <p
      className="suggestion__description"
    >
      {{description}}
    </p>
  </div>
</div>
`;

exports[`<Suggestions/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<div
  className="suggestions "
>
  <div
    className="suggestion undefined"
    key="0"
  >
    <div
      className="suggestion__label"
    >
      {{label-only}}
    </div>
  </div>
  <div
    className="suggestion undefined"
    key="1"
  >
    <div
      className="suggestion__hierarchy"
    >
      <span
        className="suggestion__hierarchy-label undefined"
        key="0"
      >
        {{root}}
      </span>
      <span
        className="suggestion__hierarchy-label undefined"
        key="1"
      >
        {{sub-children}}
      </span>
      <span
        className="suggestion__hierarchy-label undefined"
        key="2"
      >
        {{label-with-hierarchy}}
      </span>
    </div>
    <div
      className="suggestion__label"
    >
      {{label-with-hierarchy}}
    </div>
  </div>
  <div
    className="suggestion undefined"
    key="2"
  >
    <div
      className="suggestion__label"
    >
      {{label-with-description}}
    </div>
    <p
      className="suggestion__description"
    >
      {{description}}
    </p>
  </div>
</div>
`;
